controllers/files.js

```diff
@@ -223,6 +255,49 @@ const registerSharedPermission = (userId, filePath, shareToken, req) => {
     });
 };
 
+/**
+ * check whether given files can be opened in LibreOffice
+ */
+const checkIfOfficeFiles = files => {
+    if (!process.env.LIBRE_OFFICE_CLIENT_URL) {
+        logger.error('LibreOffice env is currently not defined.');
+        return;
+    }
+
+    const officeFileTypes = [
+        'application/vnd.openxmlformats-officedocument.wordprocessingml.document',     //.docx
+        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',           //.xlsx
+        'application/vnd.openxmlformats-officedocument.presentationml.presentation',   //.pptx
+        'application/vnd.ms-powerpoint',                                               //.ppt
+        'application/vnd.ms-excel',                                                    //.xlx
+        'application/vnd.ms-word',                                                     //.doc
+        'application/vnd.oasis.opendocument.text',                                     //.odt
+        'text/plain',                                                                  //.txt
+        'application/msword'                                                           //.doc
+    ];
+
+    files.forEach(f => f.isOfficeFile = officeFileTypes.indexOf(f.type) >= 0);
+};
+
+/**
+ * generates the correct LibreOffice url for (only) opening office-files
+ * @param {*} fileId, the id of the file which has to be opened in LibreOffice
+ * @param {*} accessToken, the auth token for the wopi-host later on
+ * see https://wopi.readthedocs.io/en/latest/overview.html#integration-process for further details
+ */
+const getLibreOfficeUrl = (fileId, accessToken) => {
+    if (!process.env.LIBRE_OFFICE_CLIENT_URL) {
+        logger.error('LibreOffice env is currently not defined.');
+        return;
+    }
+
+    // in the form like: http://ecs-80-158-4-11.reverse.open-telekom-cloud.com:9980/
+    const libreOfficeBaseUrl = process.env.LIBRE_OFFICE_CLIENT_URL;
+    const wopiRestUrl = process.env.BACKEND_URL || 'http://localhost:3030/';
```

Comment:
We should use `PUBLIC_BACKEND_URL` here, shouldn't we?

Reply:
you are right :+1: